woocommerce/woocommerce-gutenberg-products-block
# hook up cart items to API data in front end

Fixes #1743 

Also initial work towards #1489 

This PR hooks up the cart line items and totals to the API data in the front end. This is just the basic initial hook up to show real cart data - there will be a lot of follow up work to flesh this out more. 

In this PR I've hooked up to API data in the front end only. The editor still uses preview data. I'm not sure what the design calls for here – we could use real cart data in editor too (if the admin has stuff added to cart).

What's working:

- Cart line items: product attributes, description (rich text, not truncated yet), image, low stock warning
- Totals area: subtotal, tax
- Show empty cart content (as authored by merchant) if shopper has no items in cart

What needs more work (follow up PRs) (this is not an exhaustive list; will add issues for these): 

- ~~Correctly handle empty cart - there is an issue with empty cart component on front end~~ #1743
- ~~Tax (this may be working, need to test more)~~
- Show progress/loading spinner while cart API data is loading #1744
- Show discount on sale line items #1745
- Truncate long cart item descriptions #1590 (?) 
- Manage cart items - remove, change quantity #1746
- Handle coupons - add/remove coupon #1747
- Handle shipping #1561 
- ~~Show correct cart total (`$36717` for my `$717` worth of products!)~~ #1748
- Mobile layout may still need work #1539

#### Accessibility


### Screenshots

<img width="1741" alt="Screen Shot 2020-02-17 at 2 09 42 PM" src="https://user-images.githubusercontent.com/4167300/74616871-2881f480-518f-11ea-8a05-7607e7d11f2c.png">

### How to test the changes in this Pull Request:

1. Clone this branch.
3. Edit / add a page and add cart block. Confirm cart block works correctly in editor, and shows preview data. Publish the page.
2. As a shopper, add some items to cart.
4. View your new block-based cart on front end and confirm product details & totals are correct.
